test/jdk/com/sun/net/httpserver/bugs/TcpNoDelayNotRequired.java

```diff
@@ -0,0 +1,117 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 6968351
+ * @summary  tcp no delay not required for small payloads
+ * @library /test/lib
+ * @run main/othervm/timeout=5 -Dsun.net.httpserver.nodelay=false  TcpNoDelayNotRequired
+ */
+
+import com.sun.net.httpserver.*;
+import jdk.test.lib.net.URIBuilder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.*;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.logging.*;
+
+public class TcpNoDelayNotRequired {
+
+    public static final Logger LOGGER = Logger.getLogger("sun.net.www.protocol.http");
+
+    public static void main (String[] args) throws Exception {
+
+        java.util.logging.Handler outHandler = new StreamHandler(System.out,
+                new SimpleFormatter());
+        outHandler.setLevel(Level.FINEST);
+        LOGGER.setLevel(Level.FINEST);
+        LOGGER.addHandler(outHandler);
+
+        InetAddress loopback = InetAddress.getLoopbackAddress();
+        InetSocketAddress addr = new InetSocketAddress (loopback, 0);
+        HttpServer server = HttpServer.create (addr, 0);
+        HttpHandler handler = new Handler();
+        HttpContext ctx = server.createContext ("/test", handler);
+        HttpContext ctx2 = server.createContext ("/chunked", handler);
+        ExecutorService executor = Executors.newCachedThreadPool();
+        server.setExecutor (executor);
+        server.start ();
+
+        HttpClient client = HttpClient.newBuilder().build();
+
+        long start = System.currentTimeMillis();
+        for(int i=0;i<1000;i++) {
+            var uri = URIBuilder.newBuilder().scheme("http").loopback().port(server.getAddress().getPort()).path("/test").build();
+            var response = client.send(HttpRequest.newBuilder(uri).build(), HttpResponse.BodyHandlers.ofString());
+            if(!response.body().equals("hello")) throw new IllegalStateException("incorrect body");
+        }
+        for(int i=0;i<1000;i++) {
+            var uri = URIBuilder.newBuilder().scheme("http").loopback().port(server.getAddress().getPort()).path("/chunked").build();
+            var response = client.send(HttpRequest.newBuilder(uri).build(), HttpResponse.BodyHandlers.ofString());
+            if(!response.body().equals("hello")) throw new IllegalStateException("incorrect body");
+        }
+        long time = System.currentTimeMillis()-start;
+        System.out.println("time "+time);
+        server.stop(0);
+        executor.shutdown();
```

Comment:
Please use a try/finally block to stop the `server`. The `try` can start immediately after the `server.start()` call.

Similarly, for the `executor and the `HttpClient`, you can use a try-with-resource block to close them too.